usr/src/cmd/zdb/zdb.c

```diff
@@ -2301,11 +2301,18 @@ dump_label(const char *dev)
 
 		(void) snprintf(path, sizeof (path), "%s%s", ZFS_RDISK_ROOTD,
 		    dev);
-		if ((s = strrchr(dev, 's')) == NULL || !isdigit(*(s + 1)))
+		if (((s = strrchr(dev, 's')) == NULL &&
+		    (s = strchr(dev, 'p')) == NULL) ||
+		    !isdigit(*(s + 1)))
 			(void) strlcat(path, "s0", sizeof (path));
```

Comment:
This is only about using a vdev name like ```c0t0d0``` - the "wholedisks" pools use ```s0```, so it's mostly used at the moment, and this doesn't stop you from providing full path, or specifying ```c0t0d0s1```.